javascript/extractor/src/com/semmle/jcorn/Parser.java

```diff
@@ -847,6 +852,10 @@ else if (Identifiers.isIdentifierStart(this.fullCharCodeAtPos(), false))
     }
 
     String str = inputSubstring(start, this.pos);
+    if (seenUnderscoreNumericSeparator) {
+      str = str.replace("_", "");
+      seenUnderscoreNumericSeparator = false;
+    }
```

Comment:
Isn't it safe to just always do the `str = str.replace("_", "");`? 

Or do you keep the boolean flag because of performance? (Could a `str.contains("_")` check be an alternative then)?

Reply:
Performance. As a rule I'd like it that rarely used language features don't degrade the performance of commonly used ones.